File: AndroidApp/src/main/java/com/smp/sharedkotlin/app/presentation/MemoMapper.kt
package com.smp.sharedkotlin.app.presentation

import com.smp.sharedkotlin.app.support.Mapper
import com.smp.sharedkotlin.entity.MemoEntity


object MemoMapper : Mapper<MemoEntity, Memo> {

    override fun map(obj: MemoEntity): Memo {
        return Memo(
            obj.id,
            obj.title,
            obj.content,
            obj.lastModifiedAt,
            obj.createdAt
        )
    }

    override fun mapAll(objs: Iterable<MemoEntity>): List<Memo> {
        return objs.map { map(it) }
    }

    override fun reverse(obj: Memo): MemoEntity {
        return MemoEntity(
            obj.id,
            obj.title,
            obj.content,
            obj.lastModifiedAt,
            obj.createdAt
        )
    }

    override fun reverseAll(objs: Iterable<Memo>): List<MemoEntity> {
        return objs.map { reverse(it) }
    }

}

File: iOSApp/iOSApp/Presentation/MemoMapper.swift
import SharedCode

class MemoMapper : Mapper {
    
    func map(obj: MemoEntity) -> Memo {
        return Memo(id: obj.id, title: obj.title, content: obj.content, createdAt: obj.createdAt, lastModifiedAt: obj.lastModifiedAt)
    }
    
    
    func mapAll(objs: [MemoEntity]) -> [Memo] {
        return objs.map { map(obj: $0) }
    }
    
    func reverse(obj: Memo) -> MemoEntity {
        return MemoEntity(id: obj.id, title: obj.title, content: obj.content, createdAt: obj.createdAt, lastModifiedAt: obj.lastModifiedAt)
    }
    
    func reverseAll(objs: [Memo]) -> [MemoEntity] {
        return objs.map { reverse(obj: $0) }
    }
}
